File: 2053-kth-distinct-string-in-an-array/2053-kth-distinct-string-in-an-array.py
class Solution:
    def kthDistinct(self, arr: List[str], k: int) -> str:
        dictk = dict()
        for i in range(len(arr)):
            if arr[i] not in dictk:
                dictk[arr[i]] = 1
            else:
                val = 1
                dictk[arr[i]] += val
        for i in range(len(arr)):
            if dictk[arr[i]] == 1:
                k -= 1
            if(k==0):
                return arr[i]
        return ""
                

File: 2053-kth-distinct-string-in-an-array/2053-kth-distinct-string-in-an-array.java
class Solution {
    public String kthDistinct(String[] arr, int k) {
        Map<String, Integer> stringFreq = new HashMap<>();
        for(String str: arr) {
            stringFreq.put(str, stringFreq.getOrDefault(str, 0)+1);
        }
        for(String str: arr) {
            if (stringFreq.get(str) == 1)
                k--;
            if (k == 0)
                return str;
        }
        return "";
    }
}